### DISCUSSION TITLE
Is it possible to merge only the relationship and not the parent?

### DISCUSSION THREAD

**Discussion by jalvespinto** [sqlalchemy/sqlalchemy#10010] [Usage Questions] [answered]
On this project I use sa ORM, but I also have a mapping system to convert every ORM instance into domain models entities and then back to ORM models when I have to persist them and I am having trouble workin with a m2m relationship. 

Let's say I have the docs m2m relationship example. For the domain model mapping I would have something like:

```py
class Base(DeclarativeBase):
    pass

association_table = Table(
    "association_table",
    Base.metadata,
    Column("left_id", ForeignKey("left_table.id")),
    Column("right_id", ForeignKey("right_table.id")),
)


class Parent(Base):
    __tablename__ = "left_table"

    id: Mapped[int] = mapped_column(primary_key=True)
    name: Mapped[str] = mapped_column()
    children: Mapped[List[Child]] = relationship(secondary=association_table)


class Child(Base):
    __tablename__ = "right_table"

    id: Mapped[int] = mapped_column(primary_key=True)

class DomainParent:
    def __init__(self,id,children):
        self.id=id,
        self.children=children

def from_sa_to_domain(sa: Parent) -> DomainParent:
    return DomainParent(
        id=sa.id,
        children=[i.id for i in sa.children]
    )

def from_domain_to_sa(domain: DomainParent) -> Parent:
    return Parent(
        id=domain.id,
        children=[Child(id=i) for i in domain.children]
    )
``` 
But now, if I try to add a new DomainParent with a different `id` but the same `children` I will get a unique key violation. I guess because there is already a `Child` instance on the session with same `id`. I then tried to use `session.merge` which solves the current problem. But then If I try to add a new DomainParent with an `id` that already exists and different `name` I would want get a violation, but since I am using `session.merge` the transaction will persist and the Parent will get the new name. How can I work around this? Is it possible to specify that the merge should take place only on the `Child`?

**Comment by zzzeek** [comment, marked as answer]
you can merge directly, like

    domain.children = [session.merge(c) for c in domain.children]

is that what you're looking for?
